src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -160,8 +160,6 @@ public void iCallThePaymentNotificationApiExpectedStatusOf(int expectedStatus, S
 
         logger.info("Payment notiifcation response: {}", scenarioScopeState.response);
         JSONObject jsonObject = new JSONObject(scenarioScopeState.response);
-        scenarioScopeState.transactionId = jsonObject.getString("transactionId");
```

Comment:
Can you check whether this method is genraic and not meant for the failure( validation )case?.
if so the transaction is check might still be valid if the expected status is 202 ( async)

Reply:
I can add the assertion for transactionId not NA in "I should get transactionId in response" step.